### PR TITLE
Fixing Scroller's RepeatedSnapPoint bug: wrong snapping before first snap point

### PR DESCRIPTION
**Pull request by RBrid**
## Description
Formula for picking the snapping point of a scroll or zoom repeated snap point was incorrect. It could pick a value outside the boundaries of [Start, End].

A concrete numerical example:
A scroll repeated snap point is declared with:
  - Start: 100
  - End: 2700
  - Offset: 500
  - Interval: 600
When inertia's natural rest point is about 150, the Scroller would attempt to go to offset -100 (== Offset - Interval) and eventually stop at 0.   -100 is not a valid target as it is smaller than Start.
The valid snapping values are: 500, 1100, 1700, 2300.  With the fix, the Scroller will now snap at 500 which is the closest snapping value to 150.

The fix:
Updated the RepeatedScrollSnapPoint::CreateRestingPointExpression and RepeatedZoomSnapPoint::CreateRestingPointExpression methods to account for the situation where the natural rest point is in between the Start and first snapping values.
Had to decrease the length of variable names in the expressions to avoid hitting the limit of composition expression string length.

## How Has This Been Tested?
Tested scroll repeated snap point using existing MuxControlsTestApp test page.
Created a new test page for zoom snap points and tested same scenario for zoom repeated snap point.
Wrote 2 new unit tests that cover broken scenario for scroll and zoom snap points.

### CODE REVIEW

File: dev/Scroller/ScrollerSnapPoint.cpp [hunk 1]
```diff
@@ -681,18 +681,33 @@ winrt::ExpressionAnimation RepeatedScrollSnapPoint::CreateRestingPointExpression
...
+        L"((Abs(T.%2!s!/T.Scale-(Floor((T.%2!s!/T.Scale-P)/V)*V+P))>=Abs(T.%2!s!/T.Scale-(Ceil((T.%2!s!/T.Scale-P)/V)*V+P)))&&((Ceil((T.%2!s!/T.Scale-P)/V)*V+P)<=(%1!s!?iE:E)))?(%1!s!?((Ceil((T.%2!s!/T.Scale-P)/V)*V+P)==M?((M==P?P*T.Scale:(M-V)*T.Scale)):(Ceil((T.%2!s!/T.Scale-P)/V)*V+P)*T.Scale):(Ceil((T.%2!s!/T.Scale-P)/V)*V+P)*T.Scale):(%1!s!?((Floor((T.%2!s!/T.Scale-P)/V)*V+P)==M?(M+V<=(%1!s!?iE:E)?(M+V)*T.Scale:M*T.Scale):(T.%2!s!/T.Scale<P-0.5*V?(Ceil((T.%2!s!/T.Scale-P)/V)*V+P)*T.Scale:(Floor((T.%2!s!/T.Scale-P)/V)*V+P)*T.Scale)):(T.%2!s!/T.Scale<P-0.5*V?(Ceil((T.%2!s!/T.Scale-P)/V)*V+P)*T.Scale:(Floor((T.%2!s!/T.Scale-P)/V)*V+P)*T.Scale))",
```
Comment (StephenLPeters) [inline dev/Scroller/ScrollerSnapPoint.cpp]: >L"((Abs(T.%2!s!/T.Scale-(Floor((T.%2!s!/T.Scale-P)/V)*V+P))>=Abs(T.%2!s!/T.Scale-(Ceil((T.%2!s!/T.Scale-P)/V)*V+P)))&&((Ceil((T.%2!s!/T.Scale-P)/V)*V+P)<=(%1!s!?iE:E)))?(%1!s!?((Ceil((T.%2!s!/T.Scale-P)/V)*V+P)==M?((M==P?P*T.Scale:(M-V)*T.Scale)):(Ceil((T.%2!s!/T.Scale-P)/V)*V+P)*T.Scale):(Ceil((T.%2!s!/T.Scale-P)/V)*V+P)*T.Scale):(%1!s!?((Floor((T.%2!s!/T.Scale-P)/V)*V+P)==M?(M+V<=(%1!s!?iE:E)?(M+V)*T.Scale:M*T.Scale):(T.%2!s!/T.Scale<P-0.5*V?(Ceil((T.%2!s!/T.Scale-P)/V)*V+P)*T.Scale:(Floor((T.%2!s!/T.Scale-P)/V)*V+P)*T.Scale)):(T.%2!s!/T.Scale<P-0.5*V?(Ceil((T.%2!s!/T.Scale-P)/V)*V+P)*T.Scale:(Floor((T.%2!s!/T.Scale-P)/V)*V+P)*T.Scale))" [](start = 8, length = 647)

This thing is insane.. thank goodness there is a comment describing it

File: dev/Scroller/ScrollerSnapPoint.cpp [hunk 2]
```diff
@@ -753,7 +768,7 @@ winrt::ExpressionAnimation RepeatedScrollSnapPoint::CreateConditionalExpression(
 
     winrt::hstring isInertiaFromImpulseExpression = GetIsInertiaFromImpulseExpression(s_interactionTracker.data());
     winrt::hstring expression = StringUtil::FormatString(
-        L"((!%1!s! && it.%2!s! / it.Scale >= stt && it.%2!s! / it.Scale <= end) || (%1!s! && it.%2!s! / it.Scale >= iStt && it.%2!s! / it.Scale <= iEnd)) && (((Floor((it.%2!s! / it.Scale - fst) / itv) * itv) + fst + aRg >= it.%2!s! / it.Scale) || (((Ceil((it.%2!s! / it.Scale - fst) / itv) * itv) + fst - aRg <= it.%2!s! / it.Scale) && ((Ceil((it.%2!s! / it.Scale - fst) / itv) * itv) + fst <= (%1!s! ? iEnd : end))))",
+        L"((!%1!s!&&T.%2!s!/T.Scale>=S&&T.%2!s!/T.Scale<=E)||(%1!s!&&T.%2!s!/T.Scale>=iS&&T.%2!s!/T.Scale<=iE))&&(((Floor((T.%2!s!/T.Scale-P)/V)*V)+P+aR>=T.%2!s!/T.Scale)||(((Ceil((T.%2!s!/T.Scale-P)/V)*V)+P-aR<=T.%2!s!/T.Scale)&&((Ceil((T.%2!s!/T.Scale-P)/V)*V)+P<=(%1!s!?iE:E))))",
```
Comment (StephenLPeters) [inline dev/Scroller/ScrollerSnapPoint.cpp]: >L"((!%1!s!&&T.%2!s!/T.Scale>=S&&T.%2!s!/T.Scale<=E)||(%1!s!&&T.%2!s!/T.Scale>=iS&&T.%2!s!/T.Scale<=iE))&&(((Floor((T.%2!s!/T.Scale-P)/V)*V)+P+aR>=T.%2!s!/T.Scale)||(((Ceil((T.%2!s!/T.Scale-P)/V)*V)+P-aR<=T.%2!s!/T.Scale)&&((Ceil((T.%2!s!/T.Scale-P)/V)*V)+P<=(%1!s!?iE:E))))", [](start = 8, length = 275)

Are the changes to this expression just compression to save characters?

File: dev/Scroller/TestUI/ScrollerZoomSnapPointsPage.xaml [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+            <TextBox x:Name="txtORSnapPointStart" AutomationProperties.Name="txtORSnapPointStart" Text="1.5" VerticalAlignment="Center"
+                Grid.Row="4" Grid.Column="1" Margin="2"/>
+            <TextBlock Text="End:" VerticalAlignment="Center"
+                Grid.Row="5" Margin="2"/>
+            <TextBox x:Name="txtORSnapPointEnd" AutomationProperties.Name="txtORSnapPointEnd" Text="7.5" VerticalAlignment="Center"
+                Grid.Row="5" Grid.Column="1" Margin="2"/>
+            <Button x:Name="btnORAddSnapPoint" AutomationProperties.Name="btnORAddSnapPoint" Content="Add Snap Point" Click="BtnORAddSnapPoint_Click"
+                Grid.Row="6" Grid.ColumnSpan="2" HorizontalAlignment="Stretch" Margin="2"/>
+        </Grid>
+        -->
```
Comment (StephenLPeters) [inline dev/Scroller/TestUI/ScrollerZoomSnapPointsPage.xaml]: maybe remove?

File: dev/Scroller/ScrollerSnapPoint.cpp (hunk 2 above)
Comment (RBrid) [reply dev/Scroller/ScrollerSnapPoint.cpp]: Yes indeed.

**Pull request merged**